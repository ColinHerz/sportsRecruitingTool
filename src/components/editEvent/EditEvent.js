import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import React, { useState } from "react";

import "./editEvent.scss";

const EditEvent = props => {
	// @TODO
	// will need to change to state array w/ names
	const players = [`Player 1`, `Player 2`, `Player 3`, `Player 4`];
	const holes = (new Array(9)).fill().map((x, i) => i + 1);
	const [scores, setScores] = useState((new Array(9)).fill().map(() => {
		return {
			"Player 1": 0,
			"Player 2": 0,
			"Player 3": 0,
			"Player 4": 0
		};
	}));

	const [active, setActive] = useState(0);

	const handlePlayerScoreChange = event => {
		event.preventDefault();

		const split = event.target.id.split(`-`, 3);
		const activeHole = parseInt(split[1]);
		const player = split[2];

		const newScores = [...scores];

		newScores[activeHole][player] = parseInt(event.target.value);

		console.log(newScores);

		setScores(newScores);
	};

	const changeHole = event => {
		event.preventDefault();

		const hole = parseInt(event.target.id.split(`-`)[1]);

		setActive(hole);
	};

	return (
		<main id="edit-event">
			<h2>Event Title</h2>

			<p>Course Name <span>9 holes</span></p>

			<section>
				<div>
					{
						holes.map((hole, index) =>
							<button
								key={index}
								id={`hole-${index}`}
								onClick={changeHole}
								className={active === index ? `active`:``}
							>
								{hole}
							</button>
						)
					}
				</div>

				<ul>
					{
						players.map(player =>
							<li key={`score-${active}-${player}`}>
								<label>
									<Link to={`/profile/${player}`}>{player}</Link>
									<input
										type="number"
										id={`score-${active}-${player}`}
										value={scores[active][player]}
										onChange={handlePlayerScoreChange}
									/>
								</label>
							</li>
						)
					}
				</ul>
			</section>
		</main>
	);
};

EditEvent.propTypes = {
	user: PropTypes.object.isRequired
};

export default EditEvent;
