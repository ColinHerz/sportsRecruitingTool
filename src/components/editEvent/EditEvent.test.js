/* eslint-env jest */

// MemoryRouter needed because header contains NavLinks, which need to
// be rendered within a Router
import { MemoryRouter } from "react-router-dom";
import React from "react";
import renderer from "react-test-renderer";

import EditEvent from "./EditEvent.js";

const genProps = () => {
	return {
		user: {}
	};
};

beforeEach(() => {
	jest.resetModules();
});

describe(`Rendering`, () => {
	it(`renders correctly`, () => {
		const props = genProps();

		const component = renderer.create(
			<MemoryRouter>
				<EditEvent {...props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
