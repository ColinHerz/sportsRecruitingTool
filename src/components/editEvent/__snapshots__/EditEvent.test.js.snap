// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders correctly 1`] = `
<main
  id="edit-event"
>
  <h2>
    Event Title
  </h2>
  <p>
    Course Name 
    <span>
      9 holes
    </span>
  </p>
  <section>
    <div>
      <button
        className="active"
        id="hole-0"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className=""
        id="hole-1"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className=""
        id="hole-2"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className=""
        id="hole-3"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className=""
        id="hole-4"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className=""
        id="hole-5"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className=""
        id="hole-6"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className=""
        id="hole-7"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className=""
        id="hole-8"
        onClick={[Function]}
      >
        9
      </button>
    </div>
    <ul>
      <li>
        <label>
          <a
            href="/profile/Player 1"
            onClick={[Function]}
          >
            Player 1
          </a>
          <input
            id="score-0-Player 1"
            onChange={[Function]}
            type="number"
            value={0}
          />
        </label>
      </li>
      <li>
        <label>
          <a
            href="/profile/Player 2"
            onClick={[Function]}
          >
            Player 2
          </a>
          <input
            id="score-0-Player 2"
            onChange={[Function]}
            type="number"
            value={0}
          />
        </label>
      </li>
      <li>
        <label>
          <a
            href="/profile/Player 3"
            onClick={[Function]}
          >
            Player 3
          </a>
          <input
            id="score-0-Player 3"
            onChange={[Function]}
            type="number"
            value={0}
          />
        </label>
      </li>
      <li>
        <label>
          <a
            href="/profile/Player 4"
            onClick={[Function]}
          >
            Player 4
          </a>
          <input
            id="score-0-Player 4"
            onChange={[Function]}
            type="number"
            value={0}
          />
        </label>
      </li>
    </ul>
  </section>
</main>
`;
